feature: purchase order registry

jniles opened the pull request:
This commit leverages FilterParser for the purchase registry.  It also
implement integration tests for the `/search` route.  Finally, it
ensures that purchase orders abides by the new identifiers syntax.

Supersedes #1114.  Closes #1114.

mbayopanda committed:
Purchase order registry

mbayopanda committed:
Implement search on purchase registry

mbayopanda committed:
Modal for search in purchase order registry

mbayopanda committed:
search filter implementation

mbayopanda committed:
Service for formating filter data

mbayopanda committed:
Fix search filter format

mbayopanda committed:
Remove temporary print pdf of registry

mbayopanda committed:
Fix errors

mbayopanda committed:
Fix and enhancement of purchase order registry

mbayopanda committed:
e2e tests for purchases registry

mbayopanda committed:
fix lint error

mbayopanda committed:
fix e2e test

Commit:
refactor(uiGrid): registry dropdowns anchors

This commit refactors the ui-grids in the application to use a[href]
links for their dropdowns. Hyperlinks are more familiar to users than
triangles.  The affected grids are users, inventory list, and Billing
Services.

It also limits the height of the inventory modal to a more sensible grid
height.

Commit:
fix(filters): ensure dates are parsed

This commit fixes a bug in the filter parser library such that it
converts dates for the client into new Date() objects.

mbayopanda committed:
fix mergin conflicts

mbayopanda committed:
fix and enhancements

Commit:
refactor(purchases): implement filter parsing

This commit leverages FilterParser for the purchase registry.  It also
implement integration tests for the `/search` route.  Finally, it
ensures that purchase orders abides by the new identifiers syntax.

Supersedes #1114.  Closes ##1114.

Commit:
Merge branch 'master' into feature-purchase-order-registry

mbayopanda commented on server/controllers/finance/cash.js:
```diff
@@ -273,7 +273,7 @@ function reference(req, res, next) {
     )c WHERE c.reference = ?;`;
 
-  db.one(sql, [ ref ])
+  db.one(sql, [ ref , 'cash'])
```
What is the reason of `cash` in `db.one(sql, [ ref , 'cash'])` ?

jniles replied:
Typo - nice catch.  It should read `db.one(sql, [ref], ref, 'cash');`.  That way it will print out a nice error message if this fails.

Although .. I'm actually not sure this route is ever used.

Commit:
fix(purchases): use proper checkbox html

This commit fixes the purchase registry to use bootstrap-compliant HTML in
its search modal.

mbayopanda reviewed (commented):
LGTM

Fix just travis errors and update the branch

Commit:
Merge branch 'master' into feature-purchase-order-registry

Commit:
fix(purchase): update test references

This commit updates the test references to the new standard.
